File: test_posts_api.js
const axios = require('axios');

// 创建axios实例
const http = axios.create({
  baseURL: 'http://localhost:8000/api',
  timeout: 60000
});

// 测试获取经验贴列表
async function testGetPosts() {
  try {
    console.log('测试获取经验贴列表...');
    const response = await http.get('/posts/', {
      params: {
        category: '',
        skip: 0,
        limit: 20
      }
    });
    console.log('成功获取经验贴列表:', response.data);
  } catch (error) {
    console.error('获取经验贴列表失败:', error.response ? error.response.data : error.message);
  }
}

// 测试根路径
async function testRoot() {
  try {
    console.log('测试根路径...');
    const response = await axios.get('http://localhost:8000/');
    console.log('根路径响应:', response.data);
  } catch (error) {
    console.error('根路径测试失败:', error.message);
  }
}

// 运行测试
async function runTests() {
  await testRoot();
  await testGetPosts();
}

runTests();

File: test_posts_api.py
import requests

# 测试根路径
try:
    print('测试根路径...')
    response = requests.get('http://localhost:8000/')
    print('根路径响应:', response.json())
except Exception as e:
    print('根路径测试失败:', str(e))

# 测试获取经验贴列表
try:
    print('测试获取经验贴列表...')
    response = requests.get('http://localhost:8000/api/posts/', params={
        'category': '',
        'skip': 0,
        'limit': 20
    })
    print('状态码:', response.status_code)
    print('响应内容:', response.json())
except Exception as e:
    print('获取经验贴列表失败:', str(e))